### PR TITLE
cmd: Make changefeed sort dir visible

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 

This PR try to revert part of the work in #1784, also fix a problem when try to set data-dir by existed changefeed information.

* `sort-dir` can be read but cannot be written.
* create dir temporarily when try to find best data-dir by existed changefeed information.

### What is changed and how it works?

revert changefeed info struct and related read and write.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### Release note

- No release note

### CODE REVIEW

**Review by amyangfei — COMMENTED** [review]
Please fix ci

**Comment by ben1009** [comment]
```
[2021-07-05T12:10:10.861Z] FAIL: fileutil_test.go:68: fileUtilSuite.TestIsDirReadWritable

[2021-07-05T12:10:10.861Z] 

[2021-07-05T12:10:10.861Z] fileutil_test.go:77:

[2021-07-05T12:10:10.861Z]     c.Assert(err, check.ErrorMatches, ".*no such file or directory")

[2021-07-05T12:10:10.861Z] ... value = nil

[2021-07-05T12:10:10.861Z] ... regex string = ".*no such file or directory"

[2021-07-05T12:10:10.861Z] ... Error value is nil
```
seems we have done several fix on fix  about `sort-dir` related issues lately, do we need have a discussion or  it is already fixed, thanks

File: cdc/server.go [hunk 1]
```diff
@@ -441,12 +441,25 @@ func (s *Server) setUpDataDir(ctx context.Context) error {
 // at the moment, only consider available disk space
 func findBestDataDir(candidates []string) (result string, ok bool) {
 	var low uint64 = 0
-	for _, dir := range candidates {
+
+	checker := func(dir string) (*util.DiskInfo, error) {
+		if err := os.MkdirAll(dir, 0o755); err != nil {
+			return nil, err
+		}
+		defer os.RemoveAll(dir)
```
Comment (overvenus) [inline cdc/server.go:449]: Why remove? What if the dir is used by other services?

Comment (3AceShowHand) [reply cdc/server.go:449]: I got it.

File: cdc/server.go [hunk 2]
```diff
@@ -441,12 +441,23 @@ func (s *Server) setUpDataDir(ctx context.Context) error {
...
+		if err != nil {
+			return nil, err
+		}
+		return info, err
+	}
+
+	for _, dir := range candidates {
+		info, err := checker(dir)
 		if err != nil {
 			log.Warn("try to get disk info failed", zap.String("dir", dir), zap.Error(err))
```
Comment (amyangfei) [inline cdc/server.go:462]: The error message can be refined, such as `check the availability of dir`

rest lgtm

**Review by amyangfei — APPROVED** [review]

**Comment by amyangfei** [comment]
By the way, have we tested the `sort-dir` configured in changefeed can work as expected @3AceShowHand

**Comment by 3AceShowHand** [comment]
> By the way, have we tested the `sort-dir` configured in changefeed can work as expected @3AceShowHand

* data-dir can be set by existed ch3angefeed info, which means we can also query it
* when creating changefeed, it will fail.

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**